enclave_manager/avalon_enclave_manager/wpe/wpe_requester.py

```diff
@@ -45,67 +45,94 @@ def get_unique_verification_key(self, nonce):
         Request wrapper to get a unique id from the KME
 
         Parameters :
-            nonce - Random hex nonce generated by this WPE
+            @param nonce - Random hex nonce generated by this WPE
         Returns :
-            response - Response received from the KME which includes
-                       the public verification key which is supposed
-                       to be included in REPORTDATA by the WPE.
+            @returns result - Result received from the KME which includes the
+                              public verification key which is supposed to be
+                              included in REPORTDATA by the WPE. None, in case
+                              of failure.
         """
         json_rpc_request = self._get_request_json("kme_uid")
         json_rpc_request["params"] = {"nonce": nonce}
+        response = self._post_and_get_result(json_rpc_request)
 
-        json_request_str = json.dumps(json_rpc_request)
-        logging.debug("Request json %s", json_request_str)
-        response = self._uri_client._postmsg(json_request_str)
-        return response
+        if "result" in response:
+            return response["result"]
+        else:
+            logger.error("Could not get a unique id from the KME : {}"
+                         .format(response))
+            return None
 
     def register_wo_processor(self, attestation_report):
         """
         Request to register this WPE with the KME
 
         Parameters :
-            attestation_report - The IAS attestation report or DCAP quote
+            @param attestation_report - The IAS attestation report/DCAP quote
         Returns :
-            response - Response received from the KME about the status of
-                       registration
-
+            @returns result - Result field containing the status of the
+                              registration. None, in case of failure.
         """
         json_rpc_request = self._get_request_json("kme_reg")
-        json_rpc_request["params"] = attestation_report
-
-        json_request_str = json.dumps(json_rpc_request)
-        logging.debug("Request json %s", json_request_str)
-        response = self._uri_client._postmsg(json_request_str)
-        return response
+        json_rpc_request["params"] = {"attestation_report": attestation_report}
```

Comment:
KME request should have same format of the typical work order requests. This is temporarily fine, you please add TODO to construct KME requests here.

Reply:
Will be taken care of at the listener which would eventually make calls to the KME trusted code.